src/main/java/org/springframework/data/repository/aot/generate/MethodMetadata.java

```diff
@@ -31,17 +35,19 @@
  * Metadata about an AOT Repository method.
  *
  * @author Christoph Strobl
+ * @since 4.0
  */
 class MethodMetadata {
 
 	private final Map<String, ParameterSpec> methodArguments = new LinkedHashMap<>();
 	private final ResolvableType actualReturnType;
 	private final ResolvableType returnType;
 
-	public MethodMetadata(RepositoryInformation repositoryInformation, Method method) {
```

Comment:
Nit: I liked the visibility modifiers to indicate which methods are desired API and to differentiate between those ones, that are package-private for testing purposes.